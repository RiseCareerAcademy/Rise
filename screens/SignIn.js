import React from "react";
import {
  Container,
  Content,
  Form,
  Item,
  Label,
  Input
} from "native-base";
import { Button, StyleSheet, View } from "react-native";

export default class SignIn extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      result: null
    };
  }

  render() {
    const { navigate } = this.props.navigation;
    return (
      <Container>
        <Content>
          <Form>
            <Item stackedLabel>
              <Label>Email</Label>
              <Input />
            </Item>
            <Item stackedLabel last>
              <Label>Password</Label>
              <Input />
            </Item>
          </Form>
        </Content>
        <View>
          <Button
            color="grey"
            onPress={() => navigate("Main")}
            style={styles.greyText}
            title="Next"
          />
        </View>
      </Container>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  },
  buttonContainer: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center"
  },
  center: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "center"
  },
  buttonStyle: {
    margin: 10
  },
  greyText: {
    color: "grey"
  },
  contentContainer: {
    paddingTop: 30
  }
});
