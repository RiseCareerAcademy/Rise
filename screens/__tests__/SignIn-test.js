import 'react-native';
import React from 'react';
import {SignIn} from '../SignIn';
import renderer from 'react-test-renderer';
alert = jest.fn();

it('returns error if all fields are not filled', () => {
    const props = {
      navigation: {
        navigate: () => { }
      },
    };
    const instance = renderer.create(<SignIn {...props} />).getInstance();
    expect(instance.validate('', '')).toEqual(false);
  });
  
  it('returns error if email is less than 5 characters long', () => {
    const props = {
      navigation: {
        navigate: () => { }
      },
    };
    const instance = renderer.create(<SignIn {...props} />).getInstance();
    expect(instance.validate('1234', '')).toEqual(false);
  });
  
  it('returns error if email contains @', () => {
    const props = {
      navigation: {
        navigate: () => { }
      },
    };
    const instance = renderer.create(<SignIn {...props} />).getInstance();
    expect(instance.validate('12345', '')).toEqual(false);
  });
  
  it('returns error if email contains .', () => {
    const props = {
      navigation: {
        navigate: () => { }
      },
    };
    const instance = renderer.create(<SignIn {...props} />).getInstance();
    expect(instance.validate('12345@', '')).toEqual(false);
  });
  
  it('returns error if password length is less than 6 characters long', () => {
    const props = {
      navigation: {
        navigate: () => { }
      },
    };
    const instance = renderer.create(<SignIn {...props} />).getInstance();
    expect(instance.validate('user@email.com', 'a')).toEqual(false);
  });

  it("returns true if all cases are passed", () => {
    const props = {
      navigation: {
        navigate: () => { }
      },
      login: () => {},
    };
    const instance = renderer.create(<SignIn {...props} />).getInstance();
    expect(instance.validate('user@email.com', 'abcdef')).toEqual(true);
  });

  it("returns true if email state is changed", () => {
    const props = {
      navigation: {
        navigate: () => { }
      },
    };
    const instance = renderer.create(<SignIn {...props} />).getInstance();
    instance.handleEmail('new@email.com');
    expect(instance.state.email).toEqual('new@email.com')
  });
  
  it("returns true if password state is changed", () => {
    const props = {
      navigation: {
        navigate: () => { }
      },
    };
    const instance = renderer.create(<SignIn {...props} />).getInstance();
    instance.handlePassword('password');
    expect(instance.state.password).toEqual('password')
  });

it('renders correctly', () => {
    const props = {
        navigation: {
            navigate: () => {}
        },
    };
  const tree = renderer.create(<SignIn {...props}>Snapshot test!</SignIn>).toJSON();

  expect(tree).toMatchSnapshot();
});
