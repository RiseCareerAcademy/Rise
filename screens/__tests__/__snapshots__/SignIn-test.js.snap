// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<settings
  navigation={
    Object {
      "navigate": [Function],
    }
  }
>
  Snapshot test!
</settings>
`;
